api/database/mongooseIdPlugin.js

```diff
@@ -0,0 +1,7 @@
+export default function mongooseIdPlugin(schema, options) {
+  schema.options.toJSON.transform = function (doc, ret, options) {
+    delete ret._id; // Remove internal _id property before exposing it as JSON.
```

Comment:
Good question. I'm new to mongodb & mongoose but AFAIK `_id` is an internal property they use to identify a document which is absolutely fine. They also provide a getter for that property so when you use `product.id` it automatically returns `_id` which is great, right? Well... I want to use that property for everything except for exposing it. So, whatever API endpoint that retrieves a document, it will automatically remove the `_id` but provide the `id` field.

TL;DR: wherever we ask for a product it will return `{ id: 'some-random-id', name: 'awesome product' }` instead of `{ _id: 'some-random-id', name: 'awesome product', id: 'some-random-id' }`.